src/instruments.rs

```diff
@@ -136,12 +317,69 @@ fn resolve_template(args: &Opts) -> &str {
     }
 }
 
-fn abbrev_name(template: &str) -> Option<&'static str> {
-    match template {
+/// Return the template name abbreviation if available.
+fn abbrev_name(template_name: &str) -> Option<&str> {
+    match template_name {
         "Time Profiler" => Some("time"),
         "Allocations" => Some("alloc"),
         "File Activity" => Some("io"),
         "System Trace" => Some("sys"),
         _ => None,
     }
 }
+
+/// Profile the target binary at `binary_filepath` and write results at
+/// `trace_filepath`.
+pub(crate) fn profile_target(
+    target_filepath: &Path,
+    xctrace_tool: &XcodeInstruments,
+    app_config: &AppConfig,
+    workspace: &Workspace,
+) -> Result<PathBuf> {
+    // 1. Get the template name either from config or interactive menu
+    let template_name: String = match app_config.template_name {
+        Some(ref name) => resolve_template_name(name).to_owned(),
+        None => {
+            let template_catalog = xctrace_tool.available_templates()?;
+            select_template(&template_catalog).unwrap()
```

Comment:
So having played around with this, I have mixed feelings.

In my profiling experience, 90% of the time you probably want the `time` profiler, and then sometimes alloc/leaks, and then everything else only in very particular circumstances. I think that throwing the user into the full list of options, without guidance, is not a great user experience.

Although I do like the polished feel, I think my *personal* inclination would be to default to `time` if no argument is provided. If we really do want to show the chooser, I would make sure `Time Profiler` was selected by default, and I would try to do something to indicate that it was the recommended template.

Basically my higher-level concern is that showing the picker does not actually improve the user experience? I'm happy to have a conversation about this, though!